### PR TITLE
fix(starrocks): lower decimal operations to fp64

### PR DESCRIPTION
**Pull request by mbrobbel**
## Description
**Motivation.** The GPU expression and aggregate paths cannot consume decimal arithmetic. Today `translate_arithmetic` refuses every decimal `ARITHMETIC_EXPR` and `aggregate_call` refuses decimal `avg`, so TPC-H Q1 (`l_extendedprice*(1-l_discount)*(1+l_tax)`, three `avg`s), Q6 (`sum(l_extendedprice*l_discount)`) and every other revenue query fail to translate on the StarRocks CN.

**What changed.** Decimal operands of `ARITHMETIC_EXPR` and the arguments of decimal `sum`/`avg` are cast to FP64 (throwing cast) and the expression yields FP64; `DECIMAL_LITERAL` with precision > 18 becomes an FP64 literal; `map_scalar_type` keeps DECIMAL at precision <= 18 and maps wider decimals to FP64 (`type_mapper::fp64_type`). The lowering is **unconditional, approximate, and never cast back**: a slot the frontend declared DECIMAL can arrive as a double and differ from StarRocks in its last digits (~1e-14 relative). This is preferred over refusing the shape; a decimal-native GPU path is the real fix.

**Verified.** `cargo fmt --check`, `cargo clippy --all-targets --no-default-features -- -D warnings`, `cargo test --workspace --no-default-features` (new: `decimal_arithmetic_is_lowered_to_fp64` (precision 31 and 18, casts pinned as throwing FP64), `decimal_avg_is_lowered_to_fp64`, `decimal_sum_is_lowered_to_fp64`, `temporal_avg_is_rejected`, `wide_decimal_literal_is_lowered_to_fp64`, `type_mapper::tests::decimal_precision_boundary_is_18`). Local run on e7b1b6ec: fmt and clippy clean; starrocks-plan-translator 6 unit + 99 `translate.rs` tests, sirius-starrocks-cn 49 tests, all passing.

**Intentionally not handled.** Exchange-boundary type mismatch for FE precision <= 18 results and the scan/GROUP BY slot rewrite for precision > 18 are tracked as separate issues (see References). `AggregateFunction.output_type` on a decimal `sum`/`avg` measure stays the FE slot type over FP64-cast arguments; DuckDB's Substrait consumer ignores `output_type` and binds from the arguments, and the exchange-side mismatch is the issue above. A shared `cast_to_fp64` helper and the merge-phase guard on `aggregate_call` arrive with the two-phase aggregation PR.

## Checklist
- [x] Read CONTRIBUTING.md and ensure PR meets "reviewability" checklist
- [x] Cover changes with new or existing tests
- [ ] Document configuration changes in code and summarize in the description above
- [x] Update human and agent documentation (README.md, docs/, skills, CLAUDE.md)

## References
Refs #1109 and #1110 (introduced the decimal arithmetic and decimal avg refusals this replaces) and #1208 (Substrait add/subtract/multiply/divide/modulus aliases the lowered arithmetic resolves through), all merged. Follow-up issues: #1687, #1688.

### CODE REVIEW

**Comment by aocsa** [comment]
Pushed 91b84da2 to this branch (maintainer edits are on) with the two small things I would otherwise have asked for, so this can flip to ready without another round trip:

1. The avg refusal said "temporal avg is not supported" but fires for every non-DOUBLE, non-decimal avg. It now reads "avg is only supported where it lowers to the GPU's FP64 avg (DOUBLE and DECIMAL inputs)", same as feat/pin-table-cn.
2. Two missing tests for the boundaries this PR introduces: `wide_decimal_literal_is_lowered_to_fp64` (DECIMAL_LITERAL with precision > 18 → `Fp64` literal) and `type_mapper::tests::decimal_precision_boundary_is_18` (a unit test, since `type_mapper` is `pub(crate)`).

I also replaced the one-line body with a full description; the "Depends on" was stale (#1111 and #1208 are merged). fmt, clippy and `cargo test --workspace --no-default-features` are green locally. The exchange-boundary mismatch for precision <= 18 and the scan/GROUP BY slot rewrite are tracked in #1687 and #1688, not here.

This PR gates the two-phase aggregation work, so once you're happy please flip it to ready and I'll approve.

**Comment by aocsa** [comment]
Pushed two more commits (maintainer edits are on), both test/doc only, after a closer review pass over the four open translator PRs:

1. e7b1b6ec `test(starrocks)`: the decimal `sum` arm of `aggregate_call` had no test at all (deleting the `"sum"` match arm left the whole suite green), so `decimal_sum_is_lowered_to_fp64` pins it at precision 18. The arithmetic and avg tests now assert the casts are *throwing FP64* casts rather than "some cast", the arithmetic test also runs at precision 18 so the FP64 `output_type` override is pinned (at precision 31 `map_type_desc` already yields FP64), and `temporal_avg_is_rejected` pins the reworded refusal, which no test reached any more.
2. e5a0613a `docs(starrocks)`: the `translate_arithmetic` comment now says where the DECIMAL/FP64 mismatch actually surfaces (Copilot's second comment is right; see the thread).

Body updated: References now point at #1109/#1110 (the refusals this replaces) instead of the stale stacking parent #1111, and the aggregate `output_type` note is under "Intentionally not handled". fmt, clippy and `cargo test --workspace --no-default-features` are green locally; the four PRs merge onto dev cleanly in any order.

File: experimental/starrocks/crates/starrocks-plan-translator/src/expr_translator.rs [hunk 1]
```diff
@@ -652,24 +677,38 @@ pub(crate) fn aggregate_call(expr: &TExpr, ctx: &mut ExprContext<'_>) -> Result<
...
-    let arguments = (0..root.num_children)
+    let mut arguments = (0..root.num_children)
         .map(|_| cursor.translate_next(ctx))
         .collect::<Result<Vec<_>>>()?;
     cursor.ensure_consumed()?;
+    if decimal_result && matches!(name, "sum" | "avg") {
+        arguments = arguments
+            .into_iter()
+            .map(|input| Expression {
+                rex_type: Some(expression::RexType::Cast(Box::new(expression::Cast {
```
[the remark this answers is not in the document]
Comment (aocsa) [reply experimental/starrocks/crates/starrocks-plan-translator/src/expr_translator.rs:706]: Checked against the consumer: `SubstraitToDuckDB::TransformAggregateOp` (`substrait/src/from_substrait.cpp`, pinned submodule a7e045b) builds each measure from `function_reference` and `arguments` only and never reads `AggregateFunction.output_type`, so DuckDB binds `sum(DOUBLE)` from the FP64 argument and the DECIMAL `output_type` is inert for execution. The real mismatch is the one you describe in the second comment: anything deriving the row schema from the frontend slot types still expects DECIMAL for a precision <= 18 result. That is tracked as #1687 and intentionally not handled here.

File: experimental/starrocks/crates/starrocks-plan-translator/src/expr_translator.rs [hunk 2]
```diff
@@ -346,16 +354,33 @@ fn translate_arithmetic(
...
-        });
-    }
-    let output_type = type_mapper::map_type_desc(&node.type_, node.is_nullable.unwrap_or(true))?;
+    // Decimal arithmetic is evaluated in FP64: the Sirius GPU expression path cannot consume
+    // decimal arithmetic, and refusing it instead -- which is what this replaced -- rejects every
+    // TPC-H revenue query. The result is approximate and is NOT cast back: a project's output slot
+    // may be declared DECIMAL while the expression yields FP64, and nothing downstream can detect
+    // that, because a Substrait `ProjectRel` carries no per-column output type. Sums of money
+    // columns therefore differ from StarRocks in the last few digits (~1e-14 relative) and render
+    // as a double. A decimal-native GPU path is the real fix.
```
[the remark this answers is not in the document]
Comment (aocsa) [reply experimental/starrocks/crates/starrocks-plan-translator/src/expr_translator.rs]: Agreed; the comment now says that the emitted `Cast`/`ScalarFunction` nodes do carry FP64 and that the mismatch is with consumers that derive the row schema from the frontend's slot types (tracked in #1687). Fixed in e5a0613a.

**Review by aocsa — APPROVED** [review]
